### ISSUE TITLE
Instagram doesnt rip

### ISSUE THREAD

**Issue by grphx7** [4pr0n/ripme#573]
No matter what page I try to rip(all pages I've tried are public) it fails immediately as it fails to download the /media folder. 

The url format I send to ripme https://www.instagram.com/page-i-want-to-rip/

**Comment by cyian-1756** [comment]
Can you post the url you're trying to rip?

**Comment by grphx7** [comment]
sudo java -jar ripme.jar -u https://www.instagram.com/wired/?hl=en
Loaded file:/home/pi/ripme/ripme.jar!/rip.properties
Loaded log4j.properties
Initialized ripme v1.4.19
[+] Creating directory: ./rips/instagram_wired
Retrieving http://instagram.com/wired
Error while loading http://instagram.com/wired/media
org.jsoup.HttpStatusException: HTTP error fetching URL. Status=404, URL=http://instagram.com/wired/media
        at org.jsoup.helper.HttpConnection$Response.execute(HttpConnection.java:459)
        at org.jsoup.helper.HttpConnection$Response.execute(HttpConnection.java:434)
        at org.jsoup.helper.HttpConnection.execute(HttpConnection.java:181)        at com.rarchives.ripme.utils.Http.response(Http.java:130)
        at com.rarchives.ripme.utils.Http.getJSON(Http.java:120)
        at com.rarchives.ripme.ripper.rippers.InstagramRipper.getFirstPage(InstagramRipper.java:78)
        at com.rarchives.ripme.ripper.AbstractJSONRipper.rip(AbstractJSONRipper.java:54)
        at com.rarchives.ripme.App.rip(App.java:59)
        at com.rarchives.ripme.App.ripURL(App.java:169)
        at com.rarchives.ripme.App.handleArguments(App.java:161)
        at com.rarchives.ripme.App.main(App.java:49)
[!] Error while ripping URL https://www.instagram.com/wired/?hl=en
java.io.IOException: Failed to load http://instagram.com/wired/media after 1 attempts
        at com.rarchives.ripme.utils.Http.response(Http.java:138)
        at com.rarchives.ripme.utils.Http.getJSON(Http.java:120)
        at com.rarchives.ripme.ripper.rippers.InstagramRipper.getFirstPage(InstagramRipper.java:78)
        at com.rarchives.ripme.ripper.AbstractJSONRipper.rip(AbstractJSONRipper.java:54)
        at com.rarchives.ripme.App.rip(App.java:59)
        at com.rarchives.ripme.App.ripURL(App.java:169)
        at com.rarchives.ripme.App.handleArguments(App.java:161)
        at com.rarchives.ripme.App.main(App.java:49)
Caused by: org.jsoup.HttpStatusException: HTTP error fetching URL. Status=404, URL=http://instagram.com/wired/media
        at org.jsoup.helper.HttpConnection$Response.execute(HttpConnection.java:459)
        at org.jsoup.helper.HttpConnection$Response.execute(HttpConnection.java:434)
        at org.jsoup.helper.HttpConnection.execute(HttpConnection.java:181)        at com.rarchives.ripme.utils.Http.response(Http.java:130)
        ... 7 more

**Comment by cyian-1756** [comment]
What version of ripme are you using?

**Comment by grphx7** [comment]
1.4.19

**Comment by cyian-1756** [comment]
It's working fine on my end. I assume the reason it's failing is http://instagram.com/wired/media is 404ing. It's probably just instagram having an outage in your area. 

What happens when you visit http://instagram.com/wired in a web browser?

**Comment by grphx7** [comment]
Well I'm glad it works for you and not me oddly enough lol. Means the problem is between keyboard and chair. 

I can access the actual page and can even access the media page too. It shows a bunch of what looks like code or something

**Comment by grphx7** [comment]
Are you using the gui or cli?

**Comment by cyian-1756** [comment]
> It shows a bunch of what looks like code or something

That should be the json data. Can you run `wget -SO- http://instagram.com/wired/media` from the computer you're running ripme on and see what that returns?

**Comment by grphx7** [comment]
It outputs the contents of /media

**Comment by cyian-1756** [comment]
> Are you using the gui or cli?

I was using the GUI

> It outputs the contents of /media

Well that's odd. I don't see why ripme would 404 and wget wouldn't. I'll try to figure out what's up and get back to you later

**Comment by grphx7** [comment]
Well I don't understand how but I got it working. I had pihole running on same computer(yeah I'm running ripme on a raspberry pi. And every time I'd run wget and try to get the instagram page or the /media it would fail. I could wget other pages too. I thought I disabled pihole and still no luck. I Uninstaller pihole and now it works fine. Thanks for walking me thru the troubleshooting.

 Feel bad I submitted an issue and it turned out to be my fault. I'm new to github is there something I do since this turned out to be my fault?
